### PR TITLE
Port to winapi 0.3

### PR DESCRIPTION
**Pull request by msiglreith**
Worked around the lacking `Debug` derives using `derivative` crates and ignore the relevant winapi structs.
Most work was finding the relevant new type locations, shouldn't include any semantic changes.

### CODE REVIEW

File: src/backend/dx12/Cargo.toml [hunk 1]
```diff
@@ -19,16 +19,11 @@ name = "gfx_backend_dx12"
...
-dxguid-sys = { git = "https://github.com/msiglreith/winapi-rs.git", branch = "gfx" }
-kernel32-sys = { git = "https://github.com/msiglreith/winapi-rs.git", branch = "gfx" }
+log = "0.3"
+smallvec = "0.4"
 spirv_cross = "0.4.2"
-user32-sys = { git = "https://github.com/msiglreith/winapi-rs.git", branch = "gfx" }
-winapi = { git = "https://github.com/msiglreith/winapi-rs.git", branch = "gfx" }
+winapi = { version = "0.3", features = ["basetsd","d3d12","d3d12sdklayers","d3d12shader","d3dcommon","d3dcompiler","dxgi1_2","dxgi1_3","dxgi1_4","dxgiformat","dxgitype","minwindef","synchapi","unknwnbase","winbase","windef","winerror","winnt","winuser"] }
 winit = { version = "0.7", optional = true }
 wio = { git = "https://github.com/msiglreith/wio-rs.git", branch = "stable" }
```
Comment (kvark) [inline src/backend/dx12/Cargo.toml]: would this be the last thing in github dependencies?

File: src/backend/dx12/src/conv.rs [hunk 2]
```diff
@@ -162,7 +166,7 @@ pub fn map_topology(primitive: Primitive) -> D3D12_PRIMITIVE_TOPOLOGY {
         TriangleStrip          => D3D_PRIMITIVE_TOPOLOGY_TRIANGLESTRIP,
         TriangleStripAdjacency => D3D_PRIMITIVE_TOPOLOGY_TRIANGLESTRIP,
         PatchList(num) => { assert!(num != 0);
-            D3D_PRIMITIVE_TOPOLOGY(D3D_PRIMITIVE_TOPOLOGY_1_CONTROL_POINT_PATCHLIST.0 + (num as u32) - 1)
+            D3D_PRIMITIVE_TOPOLOGY_1_CONTROL_POINT_PATCHLIST+ (num as u32) - 1
```
Comment (kvark) [inline src/backend/dx12/src/conv.rs]: nice!
nit: let's have some space before `+`

File: src/backend/dx12/src/native.rs [hunk 3]
```diff
@@ -238,55 +259,61 @@ pub enum DescriptorRangeBinding {
...
 }
 
-#[derive(Debug)]
+#[derive(Derivative)]
+#[derivative(Debug)]
 pub struct DescriptorSet {
     // Required for binding at command buffer
-    pub(crate) heap_srv_cbv_uav: ComPtr<winapi::ID3D12DescriptorHeap>,
-    pub(crate) heap_samplers: ComPtr<winapi::ID3D12DescriptorHeap>,
+    #[derivative(Debug="ignore")]
```
Comment (kvark) [inline src/backend/dx12/src/native.rs]: all those ignores are unfortunate... hopefully, not for long...

File: src/backend/dx12/Cargo.toml (hunk 1 above)
Comment (msiglreith) [reply src/backend/dx12/Cargo.toml]: yes, iirc
retep998 mentioned to work on it soon™️ !

**Comment by msiglreith** [comment]
Oh bors was already running
bors r=kvark

**Comment by msiglreith** [comment]
bors r=kvark

**Pull request merged**